From: Eli Cohen <eli@dev.mellanox.co.il>
Date: Thu, 9 Feb 2012 18:52:50 +0200
Subject: IB/mlx4: Set bad_wr for invalid send opcode
Patch-mainline: v3.4-rc1
Git-commit: 4ba6b8eaa9d67d03fb653cb8a13afca3236d4d70
References: bnc#786036 FATE#314304

If the opcode of a work request exceeds the range of valid opcodes,
return the pointer to the offending work request.

Signed-off-by: Eli Cohen <eli@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/infiniband/hw/mlx4/qp.c |    1 +
 1 file changed, 1 insertion(+)
--- a/drivers/infiniband/hw/mlx4/qp.c
+++ b/drivers/infiniband/hw/mlx4/qp.c
@@ -1876,6 +1876,7 @@ int mlx4_ib_post_send(struct ib_qp *ibqp
 		wmb();
 
 		if (wr->opcode < 0 || wr->opcode >= ARRAY_SIZE(mlx4_ib_opcode)) {
+			*bad_wr = wr;
 			err = -EINVAL;
 			goto out;
 		}
